pkg/image/api/helper.go

```diff
@@ -168,6 +169,16 @@ func (r DockerImageReference) RepositoryName() string {
 	return r.Exact()
 }
 
+// RegistryHostname returns the registry hostname without port
+func (r DockerImageReference) RegistryHostname() string {
+	registryHost := r.AsV2().Registry
+	if strings.Contains(registryHost, ":") {
```

Comment:
so I'd say if port is not specified, it can be 443 or 80 (secure or insecure).  If it is specified, it must match.  I.e. docker.io matches docker.io:443, but docker.io does not match docker.io:8443